test-integ/catalogv2/explicit_destinations_l7_test.go

```diff
@@ -0,0 +1,486 @@
+	cfg := testSplitterFeaturesL7ExplicitDestinationsCreator{}.NewConfig(t)
+
+	sp := sprawltest.Launch(t, cfg)
+
+	var (
+		asserter = topoutil.NewAsserter(sp)
+
+		topo    = sp.Topology()
+		cluster = topo.Clusters["dc1"]
+
+		ships = topo.ComputeRelationships()
+	)
+
+	clientV2 := sp.ResourceServiceClientForCluster(cluster.Name)
+
+	t.Log(topology.RenderRelationships(ships))
+
+	// Make sure things are in v2.
+	libassert.CatalogV2ServiceHasEndpointCount(t, clientV2, "static-client", nil, 1)
+	libassert.CatalogV2ServiceHasEndpointCount(t, clientV2, "static-server-v1", nil, 1)
+	libassert.CatalogV2ServiceHasEndpointCount(t, clientV2, "static-server-v2", nil, 1)
+	libassert.CatalogV2ServiceHasEndpointCount(t, clientV2, "static-server", nil, 0)
+
+	// Check relationships
+	for _, ship := range ships {
+		t.Run("relationship: "+ship.String(), func(t *testing.T) {
+			var (
+				svc = ship.Caller
+				u   = ship.Upstream
+			)
+
+			v1ID := ship.Upstream.ID
+			v1ID.Name = "static-server-v1"
+			v1ClusterPrefix := clusterPrefix(u.PortName, v1ID, u.Cluster)
+
+			v2ID := ship.Upstream.ID
+			v2ID.Name = "static-server-v2"
+			v2ClusterPrefix := clusterPrefix(u.PortName, v2ID, u.Cluster)
+
+			switch u.PortName {
+			case "tcp":
+				// we expect 2 clusters, one for each leg of the split
+				asserter.UpstreamEndpointStatus(t, svc, v1ClusterPrefix+".", "HEALTHY", 1)
+				asserter.UpstreamEndpointStatus(t, svc, v2ClusterPrefix+".", "HEALTHY", 1)
+
+				// Both should be possible.
+				v1Expect := fmt.Sprintf("%s::%s", cluster.Name, v1ID.String())
+				v2Expect := fmt.Sprintf("%s::%s", cluster.Name, v2ID.String())
+
+				asserter.CheckBlankspaceNameTrafficSplitViaTCP(t, svc, u,
+					map[string]int{v1Expect: 10, v2Expect: 90}, 2)
+
+			case "grpc":
+				// we expect 2 clusters, one for each leg of the split
+				asserter.UpstreamEndpointStatus(t, svc, v1ClusterPrefix+".", "HEALTHY", 1)
+				asserter.UpstreamEndpointStatus(t, svc, v2ClusterPrefix+".", "HEALTHY", 1)
+
+				// Both should be possible.
+				v1Expect := fmt.Sprintf("%s::%s", cluster.Name, v1ID.String())
+				v2Expect := fmt.Sprintf("%s::%s", cluster.Name, v2ID.String())
+
+				asserter.CheckBlankspaceNameTrafficSplitViaGRPC(t, svc, u,
+					map[string]int{v1Expect: 10, v2Expect: 90}, 2)
+
+			case "http":
+				// we expect 2 clusters, one for each leg of the split
+				asserter.UpstreamEndpointStatus(t, svc, v1ClusterPrefix+".", "HEALTHY", 1)
+				asserter.UpstreamEndpointStatus(t, svc, v2ClusterPrefix+".", "HEALTHY", 1)
+
+				// Both should be possible.
+				v1Expect := fmt.Sprintf("%s::%s", cluster.Name, v1ID.String())
+				v2Expect := fmt.Sprintf("%s::%s", cluster.Name, v2ID.String())
+
+				asserter.CheckBlankspaceNameTrafficSplitViaHTTP(t, svc, u, false, "/",
+					map[string]int{v1Expect: 10, v2Expect: 90}, 2)
+			case "http2":
+				asserter.UpstreamEndpointStatus(t, svc, v1ClusterPrefix+".", "HEALTHY", 1)
+
+				// Only v1 is possible.
```

Comment:
when will this be possible in v2.  can you update the comment as to what is blocking this for v2?

Reply:
Ah yeah I can see how you would misinterpret my terrible comment here. More like "i only configured the splits to apply to the other ports" but probably the better thing to do here for symmetry is to have the http2 approach also test a 10/90 split.